### PR TITLE
Improve RPC call and add minor fixes in client

### PR DESCRIPTION
**Pull request by mille-nium**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -2,46 +2,58 @@
...
   this.active = false;
-  this.host = options.host;
-  this.port = options.port;
+  this.transportType = transport; // ws, wss, tcp, tls
+  this.options = options; // set path as option for IPC
   this.parcelId = 0;
-  const transport = options.transport; // ws, wss, tcp, tls
 
-  if (!transport) {
+  if (!this.transportType) {
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]: You can make it one-liner

File: lib/client.js [hunk 2]
```diff
@@ -2,46 +2,58 @@
...
   }
 
-  const Transport = transports[transport];
-  this.transport = new Transport({ host: this.host, port: this.port });
+  const Transport = transports[this.transportType];
+  this.transport = new Transport();
   Connection.call(this);
 
-  this.transport.on('error', (err) => {
+  this.transport.on('error', err => {
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]: Same here

File: lib/client.js [hunk 3]
```diff
@@ -83,59 +124,184 @@ Client.prototype.end = function(data, encoding) {
...
+              methodName,
+              ...args
+            );
+          };
+        });
+
+        return api;
+      };
+
+      for (const interfaceName in app) {
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]: No need for parentheses

**Comment by tshemsedinov** [comment]
Closed in favour of new version 0.1.x implementing new specification

**Pull request closed without merging**